packages/browser/src/plugins/remote-loader/index.ts

```diff
@@ -113,25 +123,41 @@ export class ActionDestination implements DestinationPlugin {
     return this.action.isLoaded()
   }
 
-  ready(): Promise<unknown> {
-    return this.action.ready ? this.action.ready() : Promise.resolve()
+  async ready(): Promise<boolean> {
+    const loadOutput = await this.loadPromise.promise
+    return !(loadOutput instanceof Error)
   }
 
   async load(ctx: Context, analytics: Analytics): Promise<unknown> {
+    if (this.loadPromise.isSettled()) {
+      const prevLoad = await this.loadPromise.promise
+      if (prevLoad instanceof Error) {
+        throw prevLoad
+      } else {
+        return prevLoad
+      }
+    }
+
     try {
       recordIntegrationMetric(ctx, {
         integrationName: this.action.name,
         methodName: 'load',
         type: 'action',
       })
-      return await this.action.load(ctx, analytics)
+
+      const loadP = this.action.load(ctx, analytics)
+
+      this.loadPromise.resolve(await loadP)
+      return loadP
```

Comment:
Why do we need to create `loadPromise` from `createDeferred`? Seems like we could set it to `this.action.load`, and then at the top of this function just return `this.loadPromise` if it exists.

Reply:
The deferred promise is used in the `ready` function. This handles a case where ready might be called before `load`